databricks/koalas/typedef/typehints.py

```diff
@@ -93,41 +93,50 @@ class NameTypeHolder(object):
 
 def as_spark_type(tpe) -> types.DataType:
     """
-    Given a python type, returns the equivalent spark type.
+    Given a Python type, returns the equivalent spark type.
     Accepts:
-    - the built-in types in python
+    - the built-in types in Python
     - the built-in types in numpy
     - list of pairs of (field_name, type)
     - dictionaries of field_name -> type
-    - python3's typing system
+    - Python3's typing system
     """
-    if tpe in (str, "str", "string"):
-        return types.StringType()
-    elif tpe in (bytes,):
+    # TODO: Add "boolean" and "string" types.
```

Comment:
I added "boolean" and "string" as TODO item since they work differently from "bool" and "str" respectively.

First, "boolean" casts the type from "bool-like values" to `boolean` whereas "bool" casts the type from various types to `bool`.

```python
>>> pd.Series([1, 2, 3]).astype("boolean")
Traceback (most recent call last):
...
TypeError: Need to pass bool-like values

>>> pd.Series([True, False, True]).astype("boolean")
0     True
1    False
2     True
dtype: boolean
```

Next, "string" casts the type from "sequence of string or pandas.NA" to `string` whereas "str" casts the type from various types to `object `.

```python
>>> pd.Series([1, 2, 3]).astype("string")
Traceback (most recent call last):
...
ValueError: StringArray requires a sequence of strings or pandas.NA

>>> pd.Series(['1', '2', '3']).astype("string")
0    1
1    2
2    3
dtype: string
```

In short, we need to add the new type `boolean` and `string` to match these behaviors.